### ISSUE TITLE
So and 1Wire not working in 3.1?

### ISSUE THREAD

**Issue by kr0815** [heishamon/HeishaMon#315]
I upgraded to 3.1 - didn´t see any 1Wire-sensors, also pulse counter counted nothing

After switching back to 3.0, it works again?

**Comment by MiG-41** [comment]
1wire worked at 3.0 , and now after update to 3.1 working as usual... 
Dont use S0...
Can you wrote more about your proble?

**Comment by geduxas** [comment]
I am on custom built but based on 3.1 no issues with 1wire.

**Comment by IgorYbema** [comment]
Should work and works for me

**Comment by kr0815** [comment]
I had an old Version of Heishamon, about 1 year old, don´t know which version

After the upgrade to 3.1 -
- no 1Wire Sensors are present
- the S0 sensor is strange, reports mucg to high values

After downgradng to 3.0

- 1Wire sensors are back as expected
- so shows also strange readings

**Comment by Messiam** [comment]
Is there a change list from 3.0 to 3.1? 
I'm still on 3.0

**Comment by geduxas** [comment]
> Is there a change list from 3.0 to 3.1? I'm still on 3.0
Only viewing PR commits.. 

Some new topic's exposed.. truely nothing special for regular use.

**Comment by IgorYbema** [comment]
You can find the changes in the PR: https://github.com/Egyras/HeishaMon/pull/310

**Comment by kr0815** [comment]
After the downgrade to 3.0  i upgraded to 3.1 again, now S0 and 1Wire are back

Also reported actual Watt loks good

What means Pulse Quality? How is it calculated?
In the Webinteface i see 
- Pulse quality: 38%
- Average pulse width: 53

In the console i get:
Pulses seen on S0 port 1: Good: 433 Bad: 1174 Average good pulse width: 60

My meter is a B&G, it has a pulse length of 90ms

My settings for min/max are 25ms / 120ms

**Comment by kr0815** [comment]
result after 1 day:
Back to 2.0 - working like expected

![image](https://user-images.githubusercontent.com/16240382/216428460-262bccd0-3b2e-4aa6-a790-d30889416df6.png)

**Comment by thenebu** [comment]
I got the same problem. I bought the Heishamon 4.7 PCB with Firmware 3.1 on it. 
Everything worked except the s0 meters. I use Eltako wsz15d-32a meters with 2000 imp and 30ms. 
When i found this Issue, I installed the 3.0 firmware. Without any changes the s0 meters worked fine. So it must be the 3.1 firmware causing the problems.

Thu Aug 24 10:22:21 2023 (3865133): Pulses seen on S0 port 2: Good: 614 Bad: 0 Average good pulse width: 32
Thu Aug 24 10:22:21 2023 (3865134): Measured Watthour on S0 port 2: 1.00
Thu Aug 24 10:22:21 2023 (3865141): Measured total Watthour on S0 port 2: 307.00
Thu Aug 24 10:22:21 2023 (3865147): Calculated Watt on S0 port 2: 896
